prov/efa/src/efa_hmem.c

```diff
@@ -85,26 +98,40 @@ static int efa_hmem_support_status_update_cuda(struct efa_hmem_support_status *c
 		return ret;
 	}
 
-	cuda_status->p2p_supported = true;
+	cuda_info->p2p_supported = true;
+	/* Eager and runting read protocols */
+	if (-FI_ENODATA != fi_param_get(&rxr_prov, "inter_max_medium_message_size", &tmp_value)) {
+		EFA_WARN(FI_LOG_DOMAIN,
+		         "The environment variable FI_EFA_INTER_MAX_MEDIUM_MESSAGE_SIZE was set, "
+		         "but EFA HMEM via Cuda API only supports eager and runting read protocols. Please "
+		         "unset the variable and run again.\n");
+		abort();
```

Comment:
I think this will cause trouble on a system that has cuda library installed, but application does not want to use CUDA.

Reply:
good point... removed abort :(